core/trino-parser/src/test/java/io/trino/sql/parser/TestSqlParser.java

```diff
@@ -1790,26 +1813,33 @@ public void testCreateTableAsSelect()
         Query querySelectColumns = simpleQuery(selectList(new Identifier("a"), new Identifier("b")), table(QualifiedName.of("t")));
         QualifiedName table = QualifiedName.of("foo");
 
-        assertStatement("CREATE TABLE foo AS SELECT * FROM t",
-                new CreateTableAsSelect(table, query, false, ImmutableList.of(), true, Optional.empty(), Optional.empty()));
-        assertStatement("CREATE TABLE foo(x) AS SELECT a FROM t",
-                new CreateTableAsSelect(table, querySelectColumn, false, ImmutableList.of(), true, Optional.of(ImmutableList.of(new Identifier("x"))), Optional.empty()));
-        assertStatement("CREATE TABLE foo(x,y) AS SELECT a,b FROM t",
+        assertThat(statement("CREATE TABLE foo AS SELECT * FROM t")).ignoringLocation().isEqualTo(
+                new CreateTableAsSelect(table, query, false, false, ImmutableList.of(), true, Optional.empty(), Optional.empty()));
+        assertThat(statement("CREATE TABLE foo(x) AS SELECT a FROM t")).ignoringLocation().isEqualTo(
+                new CreateTableAsSelect(table, querySelectColumn, false, false, ImmutableList.of(), true, Optional.of(ImmutableList.of(new Identifier("x"))), Optional.empty()));
+        assertThat(statement("CREATE TABLE foo(x,y) AS SELECT a,b FROM t")).ignoringLocation().isEqualTo(
```

Comment:
Thank you for refactoring the tests. Please extract the refactor as a preparatory commit. The main commit should be focused on adding the new syntax.